src/modules/AddOnProduct.rb

```diff
@@ -2198,6 +2205,21 @@ def y2update_path(src_id)
       log.info("Found y2update.tgz file from the installer extension package: #{y2update}")
       y2update
     end
+
+    # Determines whether the repositories libzypp information might have change
+    #
+    # This method returns true when the list of enabled repositories has changed.
+    # Disabled repositories are not even considered.
+    #
+    # @return [Boolean]
+    def libzypp_repos_changed?
+      repos = Y2Packager::Repository.all(enabled_only: true).sort_by(&:repo_id).map do |repo|
```

Comment:
I'd avoid sorting by `repo_id`, that's volatile and will change if you remove a repository and then add the same again. I'd rather sort the final URLs.

But there is a different problem, it's more complicated by using the DVD media. You might have two repositories with `dvd:///` URL but containing a completely different product/packages. So if you add a `dvd:///` repository, remove it and add it again but with a different medium in the drive you won't be able to detect the change if both use the same `/` product_dir... :worried: 

I do not know how to properly solve this, we would need to somehow digest the content of the medium (e.g. create a SHA sum of all `"<name>-<version>-<arch>"` packages on the medium or something like that and use that instead of the URL)...

Reply:
Finally, I've decided to add the `repo_id` to the set of values to check so we are able to detect the `dvd:///` problem you have noticed.